lib/chromic_pdf/pdf/connection/json_rpc.ex

```diff
@@ -59,11 +59,22 @@ defmodule ChromicPDF.Connection.JsonRPC do
 
   @spec is_response?(message(), call_id()) :: boolean()
   def is_response?(msg, call_id) do
-    Map.has_key?(msg, "result") && msg["id"] == call_id
+    (Map.has_key?(msg, "result") or Map.has_key?(msg, "error")) and
+      msg["id"] == call_id
   end
 
   @spec is_notification?(message(), method()) :: boolean()
   def is_notification?(msg, method) do
     Map.has_key?(msg, "method") && msg["method"] == method
   end
+
+  @spec is_error?(message()) :: boolean()
+  def is_error?(msg) do
+    Map.has_key?(msg, "error")
+  end
+
+  @spec extract_error(message()) :: String.t()
+  def extract_error(%{"error" => error}) do
+    "#{error["message"]} (Code #{error["code"]})"
```

Comment:
I wasn't sure about this one.
If I just pass the error Map we'll need some extra handling in ChromeError as the way it's coded now it only supports binaries.

Reply:
It's great like this :+1: IMO handling it here is better than in `ChromeError`, which doesn't know about JSONRPC